### DISCUSSION TITLE
Does #if ASYNCIFY work in user code?

### DISCUSSION THREAD

**Discussion by iliakan** [emscripten-core/emscripten#18786] [Q&A] [answered]
Is it possible to detect if ASYNCIFY is enabled in C code compiled with Emscripten?

This doesn't work:
```c
#if ASYNCIFY
...
#endif
```

**Comment by sbc100** [comment, marked as answer]
It will work in JS library files, but not in C/C++ files (this is the same for all the emscripten settings BTW).  

Further more `-sASYNCIFY` is a link-time option so it not known at compile time, if you want to have a compile flag you would just add your own `-DASYNCIFY`

**Comment by kripken** [comment]
See also [`emscripten_has_asyncify`](https://emscripten.org/docs/api_reference/emscripten.h.html?#c.emscripten_has_asyncify) for runtime detection.
